File: ObjectiveC/JWSLandscapeObjectiveC/JWSLandscapeObjectiveC/Class/Base/BaseViewController.m
//
//  BaseViewController.m
//  JWSLandscapeObjectiveC
//
//  Created by Clint on 2018. 1. 10..
//  Copyright © 2018년 clintjang. All rights reserved.
//

#import "BaseViewController.h"
#import "DataManager.h"

@interface BaseViewController ()
// Reset rotation related settings
- (void)setDefaultOrientations;
@end

@implementation BaseViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (BOOL) shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)interfaceOrientation
{
    if ([DataManager sharedManager].isEnableLandscape == YES) return YES;
    return (UIInterfaceOrientationIsPortrait(interfaceOrientation));
}

-  (UIInterfaceOrientationMask) supportedInterfaceOrientations
{
    if ([DataManager sharedManager].isEnableLandscape == YES)
    {
        if ([DataManager sharedManager].forcedLandscape == YES)
        {
            return UIInterfaceOrientationMaskLandscapeLeft | UIInterfaceOrientationMaskLandscapeRight;
        }
        return UIInterfaceOrientationMaskAll;
    }
    return UIInterfaceOrientationMaskPortrait;
}

- (IBAction)onClose:(id)sender {
    [self setDefaultOrientations];

    [self dismissViewControllerAnimated:YES completion:^{
        
    }];
}

- (void)setDefaultOrientations {
    [DataManager sharedManager].isEnableLandscape = NO;
    [DataManager sharedManager].forcedLandscape = NO;
}
@end

File: JWSLandscape/Class/Base/BaseViewController.swift
//
//  BaseViewController.swift
//  JWSLandscape
//
//  Created by Clint on 2017. 10. 17..
//  Copyright © 2017년 Clint. All rights reserved.
//

import UIKit

class BaseViewController: UIViewController {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
    }

    override var shouldAutorotate: Bool {
        return DataManger.shared.isEnableLandscape  // As an internal option if needed
    }
    
    @IBAction func onClose(_ sender: Any) {
        DataManger.shared.isEnableLandscape = false
        self.dismiss(animated: true) {
            
        }
    }
    
//    override func shouldAutomaticallyForwardRotationMethods() -> Bool {
//        return true
//    }
}

